src/renderers/shared/fiber/ReactFiberPendingWork.js

```diff
@@ -56,9 +56,26 @@ exports.findNextUnitOfWorkAtPriority = function(currentRoot : Fiber, priorityLev
       // because it is the highest priority for the whole subtree.
       // TODO: Coroutines can have work in their stateNode which is another
       // type of child that needs to be searched for work.
-      if (current.child) {
-        // Ensure we have a work in progress copy to backtrack through.
+      if (current.childInProgress) {
```

Comment:
The fibers in this set are "work in progress" fibers already so we don't need to clone them.

Since this algorithm searches "current" fibers, we can't use it search for work. Instead we do the search for work inline here.

Unfortunately there is an edge case further down that makes this a bit more complicated that I'm working on fixing now. When searching through completed work within one of these children we don't know if *those* children are "current" or "work in progress" so mutations could be on the wrong one... maybe. Searching for a solution now.